### ISSUE TITLE
Training of BUTD produced very bad results

### ISSUE THREAD

**Issue by netw0rkf10w** [facebookresearch/mmf#652]
Hello,

I tried training the BUTD model using the command provided in the [documentation](https://mmf.sh/docs/projects/butd/):

```
mmf_run config=projects/butd/configs/coco/defaults.yaml \
    model=butd \
    dataset=coco \
    run_type=train
```

I didn't change anything except two minor configuration values in `defaults.yaml`:

```
training:
  tensorboard: true
  num_workers: 10
```

The results are very very bad at the end, the best validation BLEU4 is 0.021671, which means the model didn't seem to learn anything:

`2020-10-21T07:49:06 | mmf.trainers.callbacks.logistics: progress: 50000/50000, val/coco/caption_cross_entropy: 2.0662, val/total_loss: 2.0662, val/coco/caption_bleu4: 0.0187, num_updates: 50000, epoch: 31, iterations: 50000, max_updates: 50000, val_time: 54s 586ms, best_update: 1000, best_iteration: 1000, best_val/coco/caption_bleu4: 0.021671`

TensorBoard output:

<img width="1030" alt="Screen Shot 2020-10-21 at 16 43 13" src="https://user-images.githubusercontent.com/8569515/96736094-8b7df800-13bc-11eb-9d8c-5b620ee44756.png">

Could you please help me on this?

Thank you very much in advance!

**Comment by netw0rkf10w** [comment]
I've tried running the evaluation using the provided pretrained model from the model zoo, and the results are also very bad: `val/coco/caption_bleu4: 0.0263`. It seems there's a bug in the computation of the metrics.

I also observed this warning:

`WARNING 2020-10-21T17:55:07 | py.warnings: /gpfsdswork/projects/rech/upf/ugt86fu/code/mmf/mmf/common/report.py:82: UserWarning: captions not found in report. Metrics calculation might not work as expected.`

**Comment by netw0rkf10w** [comment]
@vedanuj Could you please have a look at this issue that seems to be critical?

**Comment by netw0rkf10w** [comment]
Something is very wrong!

Running the provided pre-trained model on the validation set, I observed that the output json contains captions that start with either "a person that is standing..." or "a person sitting on..." for all the images!

**Comment by vedanuj** [comment]
What is the Bleu score you are getting after running the pretrained model on the COCO validation set?

**Comment by netw0rkf10w** [comment]
@vedanuj I obtained `0.0263`, as mentioned in my previous comment. It seems something is wrong in the decoding...

**Comment by vedanuj** [comment]
We will look into it. Thanks for reporting the issue.

**Comment by apsdehal** [comment]
We have tested this locally and it should be fixed with #664. Please use the following command to test the BUTD model:

```
CUDA_VISIBLE_DEVICES=0 mmf_run dataset=coco model=butd config=projects/butd/configs/coco/beam_search.yaml run_type=val checkpoint.resume_zoo=butd.coco
```

Make sure you run the evaluation on single GPU.
